src/shared/dropdown-list/dropdown-list.tsx

```diff
@@ -139,8 +139,8 @@ export const DropdownList = <T, V>({
     // =============================================================================
     // HELPER FUNCTIONS
     // =============================================================================
-    const getValue = (item: T): V | T => {
-        return valueExtractor ? valueExtractor(item) : item;
+    const getValue = (item: T): V => {
+        return valueExtractor ? valueExtractor(item) : (item as unknown as V);
```

Comment:
Why do we use a double assertion here? rather than do the original?

Reply:
Trying to treat the extracted value as `V` always. Since `V` could be any arbitrary type, Typescript doesn't accept `T as V`, so casting to unknown works around that